### ISSUE TITLE
ClosestMeaningAdapter nltk_data download loop

### ISSUE THREAD

**Issue by Nixellion** [gunthercox/ChatterBot#350]
Whenever I try to use ClosestMeaningAdapter, it goes into download loop like this:

```
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\nixes\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
```

However nltk is downloaded. Is it me, or is there a bug?

**Comment by gunthercox** [comment]
I've seen this before, but it appears to be somewhat sporadic. Occasionally data files will be downloaded for NLTK even when they have already been downloaded. I will look into the issue.

**Comment by Nixellion** [comment]
Thanks. Happens all the time here :(

**Comment by vkosuri** [comment]
I believe you were seeing this looping message due to https://github.com/gunthercox/ChatterBot/blob/master/chatterbot/utils/wordnet.py#L18

Usually we don't have privileges default nltk download directories, obviously We will get LookupError Exception, 

And also the expected default directories are ``/home/<user/nltk`` on ``Linux``, `` C:\Users\<user>\AppData\Roaming\nltk_data`` on ``Windows``

```
if sys.platform.startswith('win'):
    # Common locations on Windows:
    path += [
        str(r'C:\nltk_data'), str(r'D:\nltk_data'), str(r'E:\nltk_data'),
        os.path.join(sys.prefix, str('nltk_data')),
        os.path.join(sys.prefix, str('lib'), str('nltk_data')),
        os.path.join(
            os.environ.get(str('APPDATA'), str('C:\\')), str('nltk_data'))
    ]
else:
    # Common locations on UNIX & OS X:
    path += [
        str('/usr/share/nltk_data'),
        str('/usr/local/share/nltk_data'),
        str('/usr/lib/nltk_data'),
        str('/usr/local/lib/nltk_data')
    ]
```

**Comment by gunthercox** [comment]
This should be resolved by #369
